Ihor/Netflix/src/HomePage/MoviePageMain/MoviePageMain.tsx

```diff
@@ -17,104 +19,125 @@ interface IMoviePageProps {
 type MovieDetailsMainProps = IMoviePageProps &
   RouteComponentProps &
   PropsFromRedux;
-class MoviePageMain extends React.Component<MovieDetailsMainProps> {
-  handleSearchChange = ({
+
+
+const MoviePageMain = (props: MovieDetailsMainProps) => {
+  let history = useHistory();
+  let location = useLocation();
+
+  const handleSearchChange = ({
     search,
     searchBy,
   }: {
     search: string;
     searchBy: string;
   }) => {
-    const queryUrl = parse(this.props.location.search) as {
+
+    const queryUrl = parse(location.search) as {
+
       sortBy: string;
       search: string;
       searchBy: string;
     };
     const { sortBy } = queryUrl;
     const query = stringify({ ...queryUrl, search, searchBy });
-    this.props.moviesRequested(sortBy, this.props.offset, searchBy, search);
-    this.props.history.push({
+
+    props.moviesRequested({
+      offset: props.offset,
+      sortBy: sortBy,
+      searchBy: searchBy,
+      search: search,
+    });
+    history.push({
+
       pathname: "/search",
       search: query,
     });
   };
 
-  handleSortChange = (sortBy: string) => {
-    const queryUrl = parse(this.props.location.search) as {
+
+  const handleSortChange = (sortBy: string) => {
+    const queryUrl = parse(location.search) as {
+
       searchBy: string;
       search: string;
     };
     const { searchBy, search } = queryUrl;
     const query = stringify({ ...queryUrl, sortBy });
-    this.props.moviesRequested(sortBy, this.props.offset, searchBy, search);
-    this.props.history.push({
+
+    props.moviesRequested({
+      offset: props.offset,
+      sortBy: sortBy,
+      searchBy: searchBy,
+      search: search,
+    });
+    history.push({
+
       pathname: "/search",
       search: query,
     });
```

Comment:
I see repeatable logic on 28, 59 and 83 lines. Would be better to move it in separate method, so it would be much easier to read and understand. Try NOT TO REPEAT YOURSELF